### DISCUSSION TITLE
how to use :not for data attribute?

### DISCUSSION THREAD

**Discussion by binaryartifex** [tailwindlabs/tailwindcss#11822] [Help] [answered]
Im using React aria components and am currently styling a table cell. It makes full use of data attributes making it very developer friendly for tailwind styling. however ive run into an issue where i need separate styles for when the cell is both selected and hovered, vs not selected and hovered.

i can achieve what i need with the below styling, however it would be nice to be able to chain an inverse data attribute to save a few lines... something like `data-[hovered]:not(data-selected):text-primary-600` or something...

```
    "data-[hovered]:bg-primary-100",
    "data-[hovered]:text-primary-600",
    "data-[selected]:bg-primary-500",
    "data-[selected]:data-[hovered]:bg-primary-600",
    "data-[selected]:data-[hovered]:text-white",
    "data-[selected]:data-[pressed]:bg-primary-700",
    "data-[selected]:text-white",
```

**Comment by wongjn** [comment, marked as answer]
You could consider using an [arbitrary variant](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) like `data-[hovered]:[&:not([data-selected])]:text-primary-600` or abstract this variant to [a variant via a Tailwind Plugin](https://tailwindcss.com/docs/plugins#dynamic-variants).

**Reply by binaryartifex** [reply]
still gettin a hang of the variant syntax, must of been the one iteration i didn't try. cheers for that
